### PR TITLE
DEV: cmake for gtest

### PR DESCRIPTION
**Pull request by samir-nasibli**
## Description
Added using Google Test

## TODO
* add test_run.sh

### CODE REVIEW

File: utils/command_build_cmake_clib.py [hunk 1]
```diff
@@ -109,7 +109,8 @@ def run(self):
             "-DDPNP_SYCL_QUEUE_MGR_ENABLE:BOOL=" + _dpctrl_exists,
             "-DDPNP_QUEUEMGR_INCLUDE_DIR=" + _dpctrl_include_dir,
             "-DDPNP_QUEUEMGR_LIB_DIR=" + _dpctrl_library_dir,
-            "-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON"
+            "-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON",
+            "-DENABLE_TESTS=ON"
```
Comment (samir-nasibli) [inline utils/command_build_cmake_clib.py:113]: @shssf Do you want me to add gtest check for `WIN` env or switch `ENABLE_TEST` off for `WIN`?

File: dpnp/backend/CMakeLists.txt [hunk 2]
```diff
@@ -288,3 +288,7 @@ message(STATUS "CMAKE_LINKER:                    ${CMAKE_LINKER}")
 message(STATUS "CMAKE_SOURCE_DIR:                ${CMAKE_SOURCE_DIR}")
 message(STATUS "DPNP_INSTALL_PREFIX:             ${CMAKE_INSTALL_PREFIX}")
 message(STATUS "CMAKE_VERBOSE_MAKEFILE:          ${CMAKE_VERBOSE_MAKEFILE}")
+
+if(ENABLE_TESTS)
```
Comment (shssf) [inline dpnp/backend/CMakeLists.txt:292]:
```suggestion
if(DPNP_BACKEND_TESTS)
```

Comment (shssf) [reply dpnp/backend/CMakeLists.txt:292]: Also, make it as `DPNP_DEBUG_ENABLE ` and move this code to the top of the file near line 50

File: dpnp/backend/tests/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,22 @@
+find_package(GTest REQUIRED)
+
+find_package(Threads REQUIRED)
```
Comment (shssf) [inline dpnp/backend/tests/CMakeLists.txt:3]: add information messages about results of these finds (like MKL or PSTL)

File: dpnp/backend/tests/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+# Emulate autotools like make check target to build tests
+# set(CMAKE_CTEST_COMMAND ctest --progress --output-on-failure -j 4)
+
+enable_testing()
+
+include_directories(${GTEST_INCLUDE_DIR})
+link_directories(${GTEST_LIB_DIR})
+
+add_executable(test_random test_random.cpp)
```
Comment (shssf) [inline dpnp/backend/tests/CMakeLists.txt:19]:
```suggestion
add_executable(dpnpc_tests test_random.cpp)

```

File: dpnp/backend/tests/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+target_link_libraries(test_random GTest::GTest GTest::Main pthread dpnp_backend_c)
+add_test(test_random test_random)
+add_custom_target(test COMMAND ${CMAKE_CTEST_COMMAND} DEPENDS test_random)
```
Comment (shssf) [inline dpnp/backend/tests/CMakeLists.txt:22]: Do not understand why we need custom target clause

File: dpnp/backend/tests/CMakeLists.txt (hunk 3 above)
Comment (samir-nasibli) [reply dpnp/backend/tests/CMakeLists.txt:3]: `find_package(Threads REQUIRED)` calls `FindThreads` module, that can easily print finding results, raise `FATAL_ERROR`. Additionally I have added message, if lib are not found.

File: dpnp/backend/CMakeLists.txt [hunk 6]
```diff
@@ -186,6 +187,13 @@ endif()
 target_include_directories(dpnp_backend_c PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include)
 target_include_directories(dpnp_backend_c PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/src)
 
+# -----------------------------------------------------------------------------------------------
+# Testing logic...
+# -----------------------------------------------------------------------------------------------
+if(DPNP_BACKEND_TESTS)
```
Comment (shssf) [inline dpnp/backend/CMakeLists.txt:193]: What will happen is nobody provides this option at compile time?
add `option(DPNP_` with default `FALSE` as above

**Pull request merged**